osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -0,0 +1,198 @@
+                        Children = new Drawable[]
+                        {
+                            new RankingsHeader
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre,
+                                Country = { BindTarget = country },
+                                Scope = { BindTarget = scope },
+                                Ruleset = { BindTarget = ruleset }
+                            },
+                            new Container
+                            {
+                                RelativeSizeAxes = Axes.X,
+                                AutoSizeAxes = Axes.Y,
+                                Children = new Drawable[]
+                                {
+                                    contentPlaceholder = new Container
+                                    {
+                                        Anchor = Anchor.TopCentre,
+                                        Origin = Anchor.TopCentre,
+                                        AutoSizeAxes = Axes.Y,
+                                        RelativeSizeAxes = Axes.X,
+                                        Margin = new MarginPadding { Vertical = 10 }
+                                    },
+                                    loading = new DimmedLoadingLayer(),
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colour)
+        {
+            Waves.FirstWaveColour = colour.Green;
+            Waves.SecondWaveColour = colour.GreenLight;
+            Waves.ThirdWaveColour = colour.GreenDark;
+            Waves.FourthWaveColour = colour.GreenDarker;
+
+            background.Colour = OsuColour.Gray(0.1f);
+        }
+
+        protected override void LoadComplete()
+        {
+            country.BindValueChanged(_ => redraw(), true);
+            scope.BindValueChanged(_ => redraw(), true);
+            ruleset.BindValueChanged(_ => redraw(), true);
+            base.LoadComplete();
+        }
+
+        public void ShowCountry(Country requested)
+        {
+            if (requested == null)
+                return;
+
+            Show();
+
+            if (country.Value?.FlagName == requested.FlagName)
+                return;
+
+            country.Value = requested;
+        }
+
+        private void redraw()
+        {
+            scrollFlow.ScrollToStart();
+
+            loading.Show();
+
+            cancellationToken?.Cancel();
+            request?.Cancel();
+
+            cancellationToken = new CancellationTokenSource();
+
+            switch (scope.Value)
+            {
+                default:
+                    contentPlaceholder.Clear();
+                    loading.Hide();
+                    return;
```

Comment:
I don't quite understand this case. I understand `RankingsScope.Spotlights` you can't do for now (due to ppy/osu-web#5356?) but other than that, you shouldn't just display nothing.

Reply:
It exists only in the test scene(until everything is ready), so I see nothing wrong with having this as a temporary solution

Reply:
moving the `cancellationToken` construction was enough to make it clear there was no issue. adding a schedule/cancel call has no meaning and is not correct.